TinyML example on ESP-EYE (#4)
fotahub/fotahub-device-sdk-esp32: docs/getting-started/espeye_tinyml.md

## Supported targets :
Both cat face detection and human face detection can be used as is with ESP EYE board.

## Installation :

A description of the SDKs and tools that must be available on your laptop or computer, advise how to install them can be found [here](https://github.com/fotahub/fotahub-device-sdk-esp32/blob/main/README.md#installation).

# Usage :
## Get the FotaHub Device SDK for ESP : 
		
Download and uncompress the [FotaHub Device SDK](https://github.com/fotahub/fotahub-device-sdk-esp32) for ESP to a location in your machine.
		
Note: Make sure that the path to the location of the FotaHub Device SDK for ESP on your machine does not contain any spaces.
			
## Create a FotaHub product : 

Create a FotaHub product that represents your board in FotaHub as explained [here](https://github.com/fotahub/fotahub-device-sdk-esp32/blob/main/docs/fotahub/create-product.md), because product information such as its name and id will be used to upload and provide the TinyML model update as previously explained.

#   Create and run initial Cat face detection TinyML model : 

> 1. Start the Visual Studio Code and open `(File > Folder...)` 

Reviewer: ```suggestion
1. Start the Visual Studio Code and open (`File` > `Folder...`) the `cat-detection` example included in the FotaHub Device SDK for ESP32 (`<device-sdk-root>/examples/esp-eye/cat-detection`).
```

Cat face detection example included in the FotaHub Device SDK for ESP32 

> (`<device-sdk-root>/examples/huzzah32/TinyML/Cat_face_detection`) .

Reviewer: ```suggestion
```

> 2. Inside main folder, go to the `Configuration.h` , and initialize the `WIFI_STATION_SSID` and `WIFI_STATION_PASSPHRASE` constants with the credentials of the Wi-Fi network you want to connect to with your board:

Reviewer: ```suggestion
2. Go to the `Configuration.h` file in the `main` folder, and initialize the `WIFI_STATION_SSID` and `WIFI_STATION_PASSPHRASE` constants with the credentials of the Wi-Fi network you want to connect to with your board:
```

```C
#define WIFI_STATION_SSID "my-fancy-WiFi-SSID"

#define WIFI_STATION_PASSPHRASE "my-cryptic-WiFi-passphrase"
```
3. Open the DemoProductInfo.h file in the main folder, and initialize the `DEMO_PRODUCT_ID` and `DEMO_PRODUCT_NAME` constants with the id and the name of the previously created FotaHub product (`see Products > {{Your FotaHub product}} > Settings > General at Fotahub`). 

![Product credentials](E0.PNG)

Leave the `DEMO-PRODUCT-FIRMWARE-VERSION` as is for now.<br>
In case you choose a different algorithm rather than SHA256 as the binary checksum algorithm or you preferred to use a signature instead, you must adjust the following constant  `DEMO-PRODUCT-FIRMWARE-UPDATE-VERIFICATION-ALGORITHM` accordingly: 

```C

> #define DEMO_PRODUCT_ID "183d06a3-881a-462f-9daa-bfe51608f8ff"

Reviewer: ```suggestion
```
